compiler/grainformat/reformat.re

```diff
@@ -2114,396 +2930,300 @@ and print_expression =
-                } else {
-                  let printed_expression =
-                    print_expression(
-                      ~parentIsArrow=false,
-                      ~endChar=None,
-                      ~original_source,
-                      ~parent_loc,
-                      e,
-                    );
-
-                  previous_line := get_end_loc_line(e.pexp_loc);
-
-                  Doc.group(
-                    Doc.concat([
-                      stmt_leading_comment_docs,
-                      blank_line_above,
-                      Doc.group(printed_expression),
-                      line_end_comments,
-                    ]),
-                  );
-                };
-              },
-              expressions,
-            ),
-          );
-
-        // we handle the  comments after the last expression in a block
-        // specially here
-
-        let remaining_comments_in_block =
-          Walktree.get_comments_inside_location(expr.pexp_loc);
-        Walktree.remove_used_comments([], remaining_comments_in_block);
-        let blockEndCommentDocs =
-          if (List.length(remaining_comments_in_block) > 0) {
-            print_multi_comments(remaining_comments_in_block, previous_line^);
-          } else {
-            Doc.nil;
-          };
-
+      switch (expressions) {
+      | [] =>
+        // Not legal syntax so we shouldn't ever hit it, but we'll handle
+        // it just in case.
         Doc.breakableGroup(
           ~forceBreak=true,
-          Doc.concat([
-            Doc.lbrace,
-            after_brace_comments,
-            Doc.indent(Doc.concat([Doc.line, block, blockEndCommentDocs])),
-            Doc.line,
-            Doc.rbrace,
-          ]),
-        );
-      } else {
-        let (_leading_comments, trailing_comments) =
-          Walktree.partition_comments(
-            ~range=Some(parent_loc),
-            ~leading_only=false,
+          Doc.concat([Doc.lbrace, Doc.indent(Doc.line), Doc.rbrace]),
+        )
+      | _ =>
+        let get_loc = (expr: Parsetree.expression) => {
+          expr.pexp_loc;
+        };
+
+        let print_item = (expr: Parsetree.expression, comments) => {
+          Doc.group(print_expression(~original_source, ~comments, expr));
+        };
+
+        let after_brace_comments =
+          Comment_utils.get_after_brace_comments(
             expr.pexp_loc,
+            comments,
+            None,
           );
 
-        Walktree.remove_used_comments([], trailing_comments);
-
-        let expr_line = get_end_loc_line(expr.pexp_loc) + 1;
+        let cleaned_comments =
+          remove_used_comments(~all_comments=comments, after_brace_comments);
+
+        let printed_expressions =
+          block_item_iterator(
+            ~previous=Block(expr.pexp_loc),
+            ~get_loc,
+            ~print_item,
+            ~comments=cleaned_comments,
+            ~get_attribute_text=
+              expr => print_attributes(expr.pexp_attributes),
```

Comment:
introduce a `let` for this func?

Reply:
Can you tell me why?

Reply:
Just a style nit, I don't like the look of anonymous functions passed as named arguments.